packages/libraries/router/src/agent.rs

```diff
@@ -112,6 +122,20 @@ fn non_empty_string(value: Option<String>) -> Option<String> {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum AgentError {
+    #[error("unable to acquire lock for {0} in {1}, reason {2}")]
+    Lock(String, String, String),
```

Comment:
```suggestion
    #[error("unable to acquire lock: {0}")]
    Lock(String),
```

I don't think stating the struct and the method name is necessary, bc the error would include the error line position anyways.

ex:
```sh
2023-10-19T17:40:51.450438Z  ERROR panicked at 'oh no, what happened?', packages/libraries/router/src/agent.rs:201:17
```